translate: react hooks to spanish (#663)
reactjs/es.react.dev: src/content/reference/react/index.md

---

## Hooks de Referencia {/*ref-hooks*/}

> El *Refs* permite a un componente [mantener alguna información que no es utilizada para renderización](/learn/referencing-values-with-refs) como un nodo del DOM o un timeout ID. A diferencia con el estado, actualizando un *ref* no vuelve a renderizar tu componente. Los *Refs* son una “escotilla de escape” del paradigma de React. Son útiles cuando tú necesitas trabajar con sistemas no-React, semejantes como el navegador incorporado de APIs.

Reviewer: ```suggestion
Los *Refs* le permiten a un componente [mantener alguna información que no es utilizada para renderización](/learn/referencing-values-with-refs) como un nodo del DOM o un timeout ID. A diferencia con el estado, actualizando un *ref* no vuelve a renderizar tu componente. Los *Refs* son una “escotilla de escape” del paradigma de React. Son útiles cuando tú necesitas trabajar con sistemas no-React, semejantes como el navegador incorporado de APIs.
```

---

## Hooks de Efecto {/*effect-hooks*/}

El *Hook Effect* permite a un componente [conectarse y sincronizarse con sistemas externos.](/learn/synchronizing-with-effects) Esto incluye lidiar con la red de trabajo, navegación DOM, animaciones, widgets escritos utilizando una libreria UI diferente y otro código no-React.

> * [`useEffect`](/reference/react/useEffect) conecta un componente a sistema externo.

Reviewer: ```suggestion
* [`useEffect`](/reference/react/useEffect) conecta un componente a un sistema externo.
```

  // ...
```

> *Hook Effect* son una “escotilla de escape” del paradigma de React. No utilices *Effects* para guiar el flujo de los datos de tu aplicación. Si no estás interactuando con un sistema externo, [no podrías necesitar un Hook Effect.](/learn/you-might-not-need-an-effect)

Reviewer: ```suggestion
*Hook Effect* son una “escotilla de escape” del paradigma de React. No utilices *Effects* para guiar el flujo de los datos de tu aplicación. Si no estás interactuando con un sistema externo, [puede que no necesites un Hook Effect.](/learn/you-might-not-need-an-effect)
```

}
```

> Algunas veces, no podrás saltar el doble renderizado porque la pantalla actualmente necesita una actualización. En ese caso, tú puedes mejorar el rendimiento separando las actualizaciones de bloqueo que deben ser síncronas (como al escribir dentro de una entrada) de actualizaciones no bloqueadas, cuáles no necesitan bloquear la interfaz de usuario (como actualizar un gráfico).

Reviewer: ```suggestion
Algunas veces, no podrás saltar el doble renderizado porque la pantalla actualmente necesita una actualización. En ese caso, tú puedes mejorar el rendimiento separando las actualizaciones de bloqueo que deben ser síncronas (como al escribir dentro de una entrada) de actualizaciones no bloqueadas, las cuáles no necesitan bloquear la interfaz de usuario (como actualizar un gráfico).
```

Para priorizar el renderizado, usa uno de estos Hooks:

- [`useTransition`](/reference/react/useTransition) permite marcar una transición de estado como no bloqueante y permitir que otras actualizaciones la interrumpan.
- [`useDeferredValue`](/reference/react/useDeferredValue) le permite aplazar la actualización de una parte no crítica de la interfaz de usuario y dejar que otras partes se actualicen primero.

---

## Otros Hooks {/*other-hooks*/}

Estos Hooks son mayormente útiles para autores de librerías y no son comúnmente usadas en el código de la aplicación.

- [`useDebugValue`](/reference/react/useDebugValue) permite personalizar la etiqueta que React DevTools muestra para tu Hook personalizado.
- [`useId`](/reference/react/useId) permite que un componente asocie un ID único consigo mismo. Normalmente, se utiliza con las API de accesibilidad.
- [`useSyncExternalStore`](/reference/react/useSyncExternalStore) permite que un componente se subscriba a una tienda externa.

---

## Tus propios Hooks {/*your-own-hooks*/}

> Tú también puedes [definir tus propios Hooks personalizados](/learn/reusing-logic-with-custom-hooks#extracting-your-own-custom-hook-from-a-component) como funciones JavaScript.

Reviewer: ```suggestion

Tú también puedes [definir tus propios Hooks personalizados](/learn/reusing-logic-with-custom-hooks#extracting-your-own-custom-hook-from-a-component) como funciones JavaScript.
```
